test/e2e/upgrade/service/service.go

```diff
@@ -259,3 +258,31 @@ func startEndpointMonitoring(ctx context.Context, m *monitor.Monitor, svc *v1.Se
 func locateService(svc *v1.Service) string {
 	return fmt.Sprintf("ns/%s svc/%s", svc.Namespace, svc.Name)
 }
+
+// TestReachableHTTPWithMinSuccessCount tests that the given host serves HTTP on the given port for a minimum of successCount number of
+// counts at a given interval. If the service reachability fails, the counter gets reset
+func TestReachableHTTPWithMinSuccessCount(host string, port int, successCount int, interval time.Duration, timeout time.Duration) {
+	pollfn := func() (bool, error) {
+		result := e2enetwork.PokeHTTP(host, port, "/echo?msg=hello",
+			&e2enetwork.HTTPPokeParams{
+				BodyContains:   "hello",
+				RetriableCodes: []int{},
+			})
+		if result.Status == e2enetwork.HTTPSuccess {
+			return true, nil
+		}
+		return false, nil // caller can retry
+	}
+
+	for retryCount := 0; retryCount < successCount; retryCount++ {
```

Comment:
I was expecting this to be slightly different: no loop here, and

```
pollFn := ... {
  PokeHTTP
  if successful {
    consecutiveSuccess++
    return consecutiveSuccess > successCount, false
  }
  consecutiveSuccess = 0
  return false, nil
}

err := wait.PollImmediate(...)
expectNoFailure(err)
```

Reply:
We want to get X consecutive successes, and we want timeouts to cause us to exit with an error (the test should fail if we can't get X consecutive before timeout.

Reply:
if we just try `n` times, that's not going to be sufficient. it has to be with some `sleep` in between. Also, the timeOut passed to the wait function is with a single request in mind. If we are going to let that govern the `n` successes, it ought to be accounting for those many requests. The current logic let's the timeout stay as is, and wraps around each request as a whole.

Reply:
What I suggested would be part of the poll loop, so even a success would wait framework.Poll in between.

Reply:
let me pull this up in a single example:

```
var conditionSuccess int
err := wait.PollImmediate(..., func() (bool, error) {
  result := e2enetwork.PokeHTTP(host, port, "/echo?msg=hello",
    &e2enetwork.HTTPPokeParams{
      BodyContains:   "hello",
      RetriableCodes: []int{},
    })
  if result.Status == e2enetwork.HTTPSuccess {
    conditionSuccess++
    return conditionSuccess >= successCount, nil
  }
  conditionSuccess = 0
  return false, nil
})
expectNoError(err)
```

Reply:
This requires successCount consecutive successes, each spaced by framework.Poll, within duration.

Reply:
What is the motivation to inline the pollfn? Is there a gotcha here that `err` will get over-written if the poll func is not inline?